Configurable Monitoring Concept Proposal (#685)
kyma-project/community: concepts/observability-configurable-monitoring/05-concept.md

> # Concept

Author: It is never final and I would like to merge it even in an early state, so let's keep it as it is

### OTLP Push Support

**Push support will be provided via OTLP only via the central deployment**

A central collector will be part of the setup which listens for OTLP metrics. The related service port can be used for pushing custom metrics of any kind. That collector will be auto-scaled depending on resource consumption. Also, that instance will be configurable via the telemetry-operators MetricPipeline resource.

### K8S & Kyma Integration

**Based on polling via hard scrape config or annotations**

K8S infrastructure will be polled as part of the scrape agent configuration. Potential replacements coming with the ote-collector will be used if possible (like for scraping node metrics). Kyma components will be scraped via annotations like any other custom workload.

### Configurability

Telemetry Operator configures pipelines of central collector, using sane defaults with focus on value for the user. User can easily select/deselect metrics by workload/namespace and so on.

### In-Cluster backend

> Monitoring component brings config to push metrics via remotewrite to managed prometheus instances. One for system, istio and custom metrics each. There will be no further tooling like the prometheus-operator ar an alertmanager available.

Reviewer: ```suggestion
The monitoring component brings config to push metrics using remote write to managed Prometheus instances. One each for system, Istio, and custom metrics. There will be no further tooling like the Prometheus operator or an alert manager available.
```

> **Push support will be provided via OTLP only via the central deployment**

Reviewer: ```suggestion
**Push support will be provided using OTLP only with the central deployment**
```
We [don't like](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology/#terminology) "via".

> A central collector, which listens for OTLP metrics, will be part of the setup. This dedicated central collector runs independent from the pull-based daemonset as the criterias for scaling are different. The central deployment mainly will auto-scale on resource consumption related to the incoming OTLP traffic. The related service port can be used for pushing custom metrics of any kind. Also, that instance will be configurable via the telemetry-operators MetricPipeline resource. In future it might be used for other kind of signals like traces as well.

Reviewer: ```suggestion
A central collector, which listens for OTLP metrics, will be part of the setup. This dedicated central collector runs independently from the pull-based daemonset because the criteria for scaling are different. The central deployment will auto-scale on resource consumption related to the incoming OTLP traffic. The related service port can be used for pushing custom metrics of any kind. Also, that instance will be configurable with the telemetry operator's MetricPipeline resource. In the future, it might also be used for other kinds of signals (like traces).
```

- Do we really need the `mainly`?
- What do you mean with `that instance`?
- `telemetry operator's`: just guessing this is what you meant?
